cares_reinforcement_learning/algorithm/mbrl/DynaSAC.py

```diff
@@ -46,28 +40,30 @@ def __init__(
         self.critic_net = critic_network.to(self.device)
         self.target_critic_net = copy.deepcopy(self.critic_net)
 
-        self.gamma = gamma
-        self.tau = tau
+        self.gamma = config.gamma
+        self.tau = config.tau
 
-        self.num_samples = num_samples
-        self.horizon = horizon
-        self.action_num = action_num
+        self.num_samples = config.num_samples
+        self.horizon = config.horizon
+        self.action_num = self.actor_net.num_actions
```

Comment:
I think this might introduce a little bit of dependency between this and actor. Could make it harder to create custom actors.

Reply:
agreed - but having it as a parameter into the algorithm means you could put the wrong value in there that conflicts with the actor...

I am uncertain what a good approach is

Reply:
the number of actions should be the same value in both cases - reading it from the actor makes sense